repository_service_tuf_api/bootstrap.py

```diff
@@ -107,47 +96,15 @@ class TUFMetadata(BaseModel):
     signed: TUFSigned
 
 
-class SettingsKeyBody(BaseModel):
-    keytype: str
-    scheme: str
-    keyid: str
-    keyid_hash_algorithms: List[str]
-    keyval: Dict[Literal["public", "private"], str]
-
-
-class SettingsKeys(BaseModel):
-    filename: str
-    password: str
-    key: SettingsKeyBody
-
-
-class RoleSettings(BaseModel):
-    # This is the from repository-service-tuf-cli RolesKeysInput
-    expiration: int
-    num_of_keys: int
-    threshold: int
-    keys: Optional[Dict[str, SettingsKeys]]
-    offline_keys: bool
-    paths: Optional[List[str]] = None
-    number_hash_prefixes: Optional[int] = None
-
-
 class ServiceSettings(BaseModel):
     targets_base_url: str
+    number_of_delegated_bins: int = Field(gt=1, lt=16385)
+    targets_online_key: bool
 
 
 class Settings(BaseModel):
-    roles: Dict[
-        Literal[
-            Roles.ROOT.value,
-            Roles.TARGETS.value,
-            Roles.SNAPSHOT.value,
-            Roles.TIMESTAMP.value,
-            Roles.BINS.value,
-        ],
-        RoleSettings,
-    ]
-    service: ServiceSettings
+    expiration: Dict[Roles, int]
+    services: ServiceSettings
```

Comment:
You must add a custom` dict()` function to the `class Settings`.
Otherwise, the RSTUF API will raise an exception because the `payload` will content python objects (`Roles` Enum objects) in the expiration.

```
{'settings': {'expiration': {<Roles.ROOT: 'root'>: 365, <Roles.TARGETS: 'targets'>: 365, <Roles.SNAPSHOT: 'snapshot'>: 1, <Roles.TIMESTAMP: 'timestamp'>: 1, <Roles.BINS: 'bins'>: 1}, 'services': {'targets_base_url': 'http://kairo.eti.br/' ...
```

```suggestion
    services: ServiceSettings
    
    def dict(self, *args, **kwargs):
        return {
            "expiration": {r.value: e for r, e in self.expiration.items()},
            "services": self.services.dict()
        }
    
```

It will do the work using the custom for expiration
```
{'settings': {'expiration': {'root': 365, 'targets': 365, 'snapshot': 1, 'timestamp': 1, 'bins': 1}, 'services': {'targets_base_url': 'http://kairo.eti.br/',
```